### PR TITLE
Add support for FR-COR real-time data to FR_O parser

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
As mentioned in the following discussion Corsica now has real-time data.

### Discussed in https://github.com/electricitymap/electricitymap-contrib/discussions/3964

<div type='discussions-op-text'>

<sup>Originally posted by **dos934** April 14, 2022</sup>
Hi everyone,

Since 2 days, Corsica real time energy mix is available : https://opendata-corse.edf.fr/pages/page-prod_electricite_filiere_cout/

 I think it is not difficult to put in in electricity map since it is the same code as réunion opendata site : https://opendata-reunion.edf.fr/pages/page-prod_electricite_filiere_cout0/ which is already in electricity map 

Someone available to do it ?:) </div>

This PR adds the endpoints and the necessary modifications to the parser for it to work but I have not had time to test it so please take your time and do so before merging it.

### CODE REVIEW

File: parsers/FR_O.py [hunk 1]
```diff
@@ -230,17 +238,18 @@ def get_date_name(zone_key: None, target_datetime: None):
...
 sources_mapping = {
     'biomass': ['bioenergies_mwh', 'bioenergies'],
-    'hydro': ['hydraulique_mwh', 'hydraulique'],
-    'solar': ['photovoltaique_mwh', 'photovoltaique', 'photovoltaique_avec_stockage'],
+    'hydro': ['hydraulique_mwh', 'hydraulique', 'micro_hydro'],
+    'solar': ['photovoltaique_mwh', 'photovoltaique', 'photovoltaique_avec_stockage', 'photovoltaique0'],
     'wind': ['eolien_mwh', 'eolien'],
     'geothermal': ['geothermie_mwh'],
-    'gas': ['biogaz']
+    'gas': ['biogaz'],
```
Comment (VIKTORVAV99) [inline parsers/FR_O.py:245]: Didn't want to change this before double checking but `biogaz` should be mapped to `biomass` right?

File: parsers/FR_O.py [hunk 2]
```diff
@@ -230,17 +238,18 @@ def get_date_name(zone_key: None, target_datetime: None):
 # Non-thermal sources names in api
 sources_mapping = {
     'biomass': ['bioenergies_mwh', 'bioenergies'],
-    'hydro': ['hydraulique_mwh', 'hydraulique'],
-    'solar': ['photovoltaique_mwh', 'photovoltaique', 'photovoltaique_avec_stockage'],
+    'hydro': ['hydraulique_mwh', 'hydraulique', 'micro_hydro'],
+    'solar': ['photovoltaique_mwh', 'photovoltaique', 'photovoltaique_avec_stockage', 'photovoltaique0'],
```
Comment (VIKTORVAV99) [inline parsers/FR_O.py:242]: `photovoltaique0` is likely a spelling error but it is in the API

File: parsers/FR_O.py (hunk 1 above)
Comment (madsnedergaard) [reply parsers/FR_O.py:245]: I think so, right @pierresegonne?

File: parsers/FR_O.py [hunk 3]
```diff
@@ -230,17 +238,18 @@ def get_date_name(zone_key: None, target_datetime: None):
...
-    'gas': ['biogaz']
+    'gas': ['biogaz'],
+    'oil': ['moteur_diesel']
 }
 
 
 def fetch_production(zone_key=None, session=None, target_datetime=None,
                      logger=logging.getLogger(__name__)):
-    if (target_datetime is None) and (zone_key != 'RE'):
+    if (target_datetime is None) and (zone_key != 'RE' or 'FR-COR'):
```
Comment (madsnedergaard) [inline parsers/FR_O.py:252]: This check is currently failing, so I'd suggest the following instead:
```suggestion
    if (target_datetime is None) and zone_key not in REAL_TIME_SOURCES.keys():
```

File: parsers/FR_O.py (hunk 1 above)
Comment (pierresegonne) [reply parsers/FR_O.py:245]: Correct

**Review by madsnedergaard — APPROVED** [review]
👏

**Pull request merged**